### PR TITLE
Use FSTConfiguration#deriveConfiguration to preserve ConfType

### LINKED ISSUE #2658: FstCodec copy constructor doesn’t preserve ConfType of FSTConfiguration
We have the following code:
```java
final Config config = new Config();
config.setCodec(new FstCodec(FSTConfiguration.createJsonConfiguration()));
final RedissonClient client = Redisson.create(config);
```
When [a copy of the codec is created by `RedisExecutor`](https://github.com/redisson/redisson/blob/6cd9b28d9cf17a5d951b28b096634f2d7171fa3b/redisson/src/main/java/org/redisson/command/RedisExecutor.java#L687), the [`ConfType` of the `FstConfiguration`](https://github.com/RuedigerMoeller/fast-serialization/blob/59f8f800b79479d0b9c80901cf55903d9b027f26/src/main/java/org/nustaq/serialization/FSTConfiguration.java#L63) is always `DEFAULT`.

This can be fixed by using [`FSTConfiguration#deriveConfiguration`](https://github.com/RuedigerMoeller/fast-serialization/blob/59f8f800b79479d0b9c80901cf55903d9b027f26/src/main/java/org/nustaq/serialization/FSTConfiguration.java#L1302).

### PR DESCRIPTION
**Pull request by eager**
fixes #2658

### CODE REVIEW

File: redisson/src/main/java/org/redisson/codec/FstCodec.java [hunk 1]
```diff
@@ -191,7 +191,7 @@ public FstCodec(ClassLoader classLoader, FstCodec codec) {
     }
 
     private static FSTConfiguration copy(ClassLoader classLoader, FstCodec codec) {
-        FSTConfiguration def = FSTConfiguration.createDefaultConfiguration();
+        FSTConfiguration def = codec.getConfig().deriveConfiguration();
```
Comment (mrniko) [inline redisson/src/main/java/org/redisson/codec/FstCodec.java]: please use `codec.config.deriveConfiguration()` instead

File: redisson/src/main/java/org/redisson/codec/FstCodec.java [hunk 2]
```diff
@@ -304,4 +304,7 @@ public ClassLoader getClassLoader() {
         return super.getClassLoader();
     }
 
+    private FSTConfiguration getConfig() {
```
Comment (mrniko) [inline redisson/src/main/java/org/redisson/codec/FstCodec.java]: This change is not needed.

File: redisson/src/main/java/org/redisson/codec/FstCodec.java (hunk 1 above)
Comment (eager) [reply redisson/src/main/java/org/redisson/codec/FstCodec.java]: Oops—not sure why I thought I needed the method 🤦‍♂ .

**Comment by mrniko** [comment]
Thank you!

**Pull request merged**